sql/pre_install/tables.sql

```diff
@@ -47,13 +47,14 @@ CREATE TABLE IF NOT EXISTS _timescaledb_catalog.hypertable (
   chunk_sizing_func_schema name NOT NULL,
   chunk_sizing_func_name name NOT NULL,
   chunk_target_size bigint NOT NULL CHECK (chunk_target_size >= 0), -- size in bytes
-  compressed boolean NOT NULL DEFAULT FALSE,
+  compression_state smallint NOT NULL DEFAULT 0,
   compressed_hypertable_id integer REFERENCES _timescaledb_catalog.hypertable (id),
   replication_factor smallint NULL CHECK (replication_factor > 0),
   UNIQUE (associated_schema_name, associated_table_prefix),
   CONSTRAINT hypertable_table_name_schema_name_key UNIQUE (table_name, schema_name),
-  CONSTRAINT hypertable_dim_compress_check CHECK (num_dimensions > 0 OR compressed = TRUE),
-  CONSTRAINT hypertable_compress_check CHECK (compressed = FALSE OR (compressed = TRUE AND compressed_hypertable_id IS NULL))
+----internal compressed hypertables have compression state = 1
```

Comment:
It would be more clear if this comment included all states:

```suggestion
-- Compression state is as follows:
-- compression_state = 0 --> hypertable or distributed hypertable with compression disabled, compressed_hypertable_id should be NULL.
-- compression_state = 1 --> internal compression hypertable that stores compressed data, compressed_hypertable_id should be NULL.
-- compression_state = 2 --> hypertable or distributed hypertable with compression enabled, compressed_hypertable_id is defined for local hypertables but not distributed hypertables
```

Ideally, I would change the order of state 1 and 2, although the exact ordinal number is not very important. I do think it is clearer that 0 and 1 is compression on/off on a non-internal hypertable. Otherwise, users will see this value go from 0 to 2 when enabling compression. Better to make the internal table the odd one.

Reply:
I initially had the states listed as you describe. But I had a really hard time getting the update script working correctly. Keeping compression_state = 1, retains the original values for "compressed" column on update and only adds a new value for the new state.

Reply:
Probably another reason why in-table modifications in the update script isn't a good idea. It would be better, IMO, to recreate the table and insert data from the old table the way we want it in the new one.